Fix msvc annoyances

adrido opened the pull request:
Some annoyances where introduced lately, and some did exist longer.
I hope its clear what every commit is doing.

sfan5 commented on src/threading/thread.cpp:
```diff
@@ -311,7 +311,7 @@ bool Thread::bindToProcessor(unsigned int proc_number)
 bool Thread::setPriority(int prio)
 {
-#if USE_WIN_THREADS
+#ifdef _WIN32
```
this is absolutely not how this should be done

change the header file at the point where decision making is done

nerzhul replied:
yeah i forgot and it seems mingw travis build doesn't care... :o

adrido replied:
soo... how should I handle it?

sfan5 replied:
see whether the `#if USE_WIN_THREADS` conditions need to be renamed or the code bock removed in this and the other function

adrido replied:
Should be done now. 
the remaining failure seems to be a failure of travis itself

sfan5 commented on src/CMakeLists.txt:
```diff
@@ -708,9 +708,12 @@ include(CheckCXXCompilerFlag)
 if(MSVC)
 	# Visual Studio
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /std:c++11")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /D WIN32_LEAN_AND_MEAN /MP")
```
don't you need to do this for mingw too then?

nerzhul commented on src/threading/thread.cpp:
```diff
@@ -311,7 +311,7 @@ bool Thread::bindToProcessor(unsigned int proc_number)
 bool Thread::setPriority(int prio)
 {
-#if USE_WIN_THREADS
+#ifdef _MSC_VER
```
use _WIN32 instead for both cases

adrido replied:
its not compatible with mingw

sfan5 replied:
Why not?

adrido replied:
If I remember correctly because it couldn't cast the return value of `getThreadHandle()` to a `HANDLE *` (`VOID *`). MSVC seems to have either a overload or an implicit cast.

adrido replied:
I added a special case for mingw.
Btw: That functions are currently not used, wouldn't it be better to completely remove them? If they are needed some day, they does still exist in git vcs.

sfan5 replied:
Looks fine to me now.

adrido commented:
Rebased and added another fix that prevented msvc from building: https://github.com/minetest/minetest/pull/5963/commits/1284717d83544142212101d446b85cd5ea0a9183

nerzhul commented:
nice @adrido .
@sfan5 can we merge ?

adrido commented:
Thanks
